src/rgw/rgw_rest_pubsub.cc

```diff
@@ -456,10 +457,10 @@ class RGWPSCreateNotif_ObjStore_S3 : public RGWPSCreateNotifOp {
     }
     try {
       // NotificationConfigurations is mandatory
+      // It can be empty which means we delete all the notifications
       RGWXMLDecoder::decode_xml("NotificationConfiguration", configurations, &parser, true);
     } catch (RGWXMLDecoder::err& err) {
-      ldpp_dout(this, 1) << "failed to parse XML payload. error: " << err << dendl;
-      return -ERR_MALFORMED_XML;
+      list_is_empty = true;
```

Comment:
not any XML parsing error is an indication that there was an empty list of configurations.
please change the implementation of `decode_xml` of `rgw_pubsub_s3_notifications` so that it won't throw an exception if the list is empty.
instead, we can just check: `configurations.list.empty()`

Reply:
My thinking here was: [Here](https://github.com/ceph/ceph/blob/master/src/rgw/rgw_pubsub.cc#L241) this error will be thrown if we don't have at least one TopicConfiguration which ultimately means list is empty. So, instead of taking this error into consideration (as we know list is empty) and doing return we can just have a boolean and consider this as an empty list?

Reply:
but if the list is not empty, but the XML is malformed, we do want to handle the exception.
just remove lines 240-242 from: rgw_pubsub.cc
if the list is empty, we **won't** get an exception. if the XML is malformed we **will** get an exception.

Reply:
Ohh yes, correct. I'll make those changes.